packages/media-utils/src/utils/validate-mime-type-for-user.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { UploadError } from './upload-error';
+import { getMimeTypesArray } from './get-mime-types-array';
+
+/**
+ * Verifies if the user is allowed to upload this mime type.
+ *
+ * @param file               File object.
+ * @param wpAllowedMimeTypes List of allowed mime types and file extensions.
+ */
+export function validateMimeTypeForUser(
+	file: File,
+	wpAllowedMimeTypes?: Record< string, string > | null
+) {
+	// Allowed types for the current WP_User.
+	const allowedMimeTypesForUser = getMimeTypesArray( wpAllowedMimeTypes );
+
+	if ( ! allowedMimeTypesForUser ) {
+		return;
+	}
+
+	const isAllowedMimeTypeForUser = allowedMimeTypesForUser.includes(
+		file.type
+	);
+
+	if ( file.type && ! isAllowedMimeTypeForUser ) {
```

Comment:
Is there a case where `file.type` won't be defined?

Reply:
It could be an empty string